gemma-core/src/main/java/ubic/gemma/core/search/source/CompassSearchSource.java

```diff
@@ -289,7 +289,7 @@ private <T extends Identifiable> Set<SearchResult<T>> performSearch( SearchSetti
             }
 
             //noinspection unchecked
-            results.add( SearchResult.from( clazz, ( T ) resultObject, score * CompassSearchSource.COMPASS_HIT_SCORE_PENALTY_FACTOR, ht, source ) );
+            results.add( SearchResult.from( clazz, ( ( T ) resultObject ).getId(), score * CompassSearchSource.COMPASS_HIT_SCORE_PENALTY_FACTOR, ht, source ) );
```

Comment:
Compass "entity" results are outdated and missing most associations, so we refetch them in the search service.